cvat-sdk/cvat_sdk/core/proxies/tasks.py

```diff
@@ -132,27 +132,25 @@ def upload_data(
             self._client.logger.info("Awaiting for task %s creation...", self.id)
             while True:
                 sleep(status_check_period)
-                (status, response) = self.api.retrieve_status(self.id)
+                requests, response = self._client.api_client.requests_api.list(
+                    action="create", task_id=self.id
+                )
+                assert response.status == 200
+                assert 1 == len(requests.results)
```

Comment:
It feels like the task data uploading endpoint could also return the rq_id for future status checks, as there is no response body at the moment. The pattern used here looks a little bit unreliable, as the request made above can fail and some other uploading request can be returned here instead.